src/Utility/Queue.hpp

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+namespace Utility {
+    /// Queue container class.
+    template<typename T> class Queue {
```

Comment:
The class allocates memory but doesn't have a destructor. What happens if the queue is deleted while it isn't empty? (Potential memory leak depending on how the queue is used.) At the very least the documentation for the class should specify that the queue has to be empty before it is deleted.

Reply:
Destructor added